src/sas/sascalc/poresize/SizeDistribution.py

```diff
@@ -79,22 +83,37 @@ def background_fit(data, power=None, qmin=None, qmax=None, type="fixed"):
     fx[idx] = data.y[idx]
 
     ##Get values of scale and if required power
-    if power is not None and power != 0:
-        # Linearize the data for a power law fit (log, log)
-        linearized_data = Data1D(np.log(data.x[idx]), np.log(fx[idx]), dy=(1/np.log(10))*sigma[idx]/fx[idx])
+    if power is not None:
+        # Linearize the data for a linear or power law fit
+        
+        linearized_data = Data1D(np.log(data.x[idx]), np.log(fx[idx]), dy=sigma[idx]/fx[idx])
+        fit_func = lambda x,b:line_func(x, power, b)
+        init_guess = (linearized_data.y[0])
+
     else:
-        linearized_data = Data1D(data.x[idx], fx[idx], dy=None)
+        # Fit both the power and scale 
+        linearized_data = Data1D(np.log(data.x[idx]), np.log(fx[idx]), dy=sigma[idx]/fx[idx])
+        fit_func = line_func
+        init_guess = (4, linearized_data.y[0])
+
+    param_result, pcov = optimize.curve_fit(fit_func, linearized_data.x, linearized_data.y, init_guess, sigma = linearized_data.dy)
+    param_err = np.sqrt(np.diag(pcov))
 
+    if len(param_err) > 1: 
+        param_err[1] = np.exp(param_result[1])*param_err[1]
+    else:
+        param_err[0] = np.exp(param_result[0])*param_err[0]
```

Comment:
If the numpy error for the parameter error has at least two elements, only modify the 2nd element, otherwise only modify the first element? Why is this happening? Does the consumer of this error array know what to do with it?

Please add inline documentation for why this is necessary.